src/main/kotlin/nexstep/mission/racingcar/RacingGame.kt

```diff
@@ -0,0 +1,22 @@
+package nexstep.mission.racingcar
+
+class RacingGame(val racingCars: List<RacingCar>) {
+    constructor(count: Int) : this(createRacingCars(count))
+
+    fun race(strategy: () -> Int): List<RacingCar> =
+        racingCars.onEach { it.move(strategy.invoke()) }
+
+    companion object {
+        private tailrec fun createRacingCars(
+            count: Int,
+            racingCars: MutableList<RacingCar> = mutableListOf()
+        ): List<RacingCar> =
+            when (true) {
+                (count == 0) -> racingCars.toList()
+                else -> {
+                    racingCars += RacingCar()
+                    createRacingCars(count - 1, racingCars)
+                }
+            }
```

Comment:
와.. 이것까지 꼬리재귀 최고입니다!

그런데 말이죠. 가변 컬렉션을 어떻게해야 불변으로 변경할 수 있을까요? 🤔

Reply:
한가지 질문이 있어요~
불변으로 변경해야하는 이유가 무엇일까요?

불변이 가지는 장점은 충분히 이해하는데요, 위 함수의 가변 컬렉션은 생명주기가 함수 내부가 전부이기에 가변이어도 괜찮지 않나 라고 생각이 들어요!

Reply:
성훈님이 말씀하신 내용도 동감합니다.

제가 가변 컬렉션을 불변으로 변경할 수 있는 부분에 대해 미션을 드린 이유는
-  현재 코드에서 결국에는 리턴 타입이 불변 리스트이기 때문이였어요. 
- 코틀린 처럼 가변과 불변이 확실한 언어가 없기도하고 가변과 불변을 잘 사용한다면, 함수도 좀 더 짧게 짤 수 있다고 생각했어요. 코틀린에 있는 수 많은 함수들을 사용하면서 라인을 줄이는 것 하나의 재미이구요 😊

아마 줄이는 과정 속에서 또 좋은 방향의 코드가 나올 수도 있지 않을까? 생각해요!

```suggestion
        private tailrec fun createRacingCars(
            count: Int,
            racingCars: List<RacingCar> = listOf()
        ): List<RacingCar> =
            when (true) {
                (count == 0) -> racingCars.toList()
                else -> {
                    createRacingCars(count - 1, racingCars + RacingCar())
                }
            }
```

아마 불변으로 만들면 이렇게 되지 않을까 합니다 🤩

Reply:
아... +연산자를 쓰면 불변 리스트를 반환하는군요... 완전 몰랐던 내용입니다 :)
의견주신 방향대로 리펙토링 해볼게요!